File: src/main/java/com/staymate/service/NoticeService.java
package com.staymate.service;

import java.util.List;

import com.staymate.entity.Notice;

public interface NoticeService {

    Notice saveNotice(Notice notice);

    List<Notice> getAllNotices();

    Notice getNoticeById(Integer noticeId);

    Notice updateNotice(Integer noticeId, Notice notice);

    void deleteNotice(Integer noticeId);

}

File: frontend/src/services/noticeService.js
import api from "./api";

export const getNotices = () =>
    api.get("/notices");

export const getNoticeById = (id) =>
    api.get(`/notices/${id}`);

export const addNotice = (notice) =>
    api.post("/notices", notice);

export const updateNotice = (id, notice) =>
    api.put(`/notices/${id}`, notice);

export const deleteNotice = (id) =>
    api.delete(`/notices/${id}`);